### ISSUE TITLE
pytorch fails

### ISSUE THREAD

**Issue by tbenst** [DavHau/mach-nix#16]
I tried,

```nix
let
  mach-nix = import (builtins.fetchGit {
    url = "https://github.com/DavHau/mach-nix/";
    ref = "master";
    rev = "42a1daf1979bef3a26a8a4eb19db801284758ba6";
  });
in
mach-nix.mkPythonShell {
  requirements = ''
    torch
  '';
}
```
<details>
  <summary>but got this error (click to expand)</summary>

    these derivations will be built:
    /nix/store/gag7dhx91bzp6i2rxc0ql1cmd78h8kqn-python3.7-torch-0.1.2.post2.drv
    /nix/store/dlm02884w3fss89ppcrbli5vm79d7dn1-python3-3.7.5-env.drv
    building '/nix/store/gag7dhx91bzp6i2rxc0ql1cmd78h8kqn-python3.7-torch-0.1.2.post2.drv'...
    Sourcing python-catch-conflicts-hook.sh
    Sourcing python-remove-bin-bytecode-hook.sh
    Sourcing setuptools-build-hook
    Using setuptoolsBuildPhase
    Using setuptoolsShellHook
    Sourcing pip-install-hook
    Using pipInstallPhase
    Sourcing python-imports-check-hook.sh
    Using pythonImportsCheckPhase
    unpacking sources
    unpacking source archive /nix/store/avcbiihlwlqdqyjqq0bn12nx88qahlx3-torch-0.1.2.post2.tar.gz
    source root is torch-0.1.2.post2
    setting SOURCE_DATE_EPOCH to timestamp 1561154629 of file torch-0.1.2.post2/setup.cfg
    patching sources
    configuring
    no configure script, doing nothing
    building
    Executing setuptoolsBuildPhase
    running bdist_wheel
    running build
    running build_deps
    Traceback (most recent call last):
    File "nix_run_setup", line 8, in <module>
        exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
    File "setup.py", line 265, in <module>
        description="Tensors and Dynamic neural networks in Python with strong GPU acceleration",
    File "/nix/store/p4xpcwj9bxrjz9zsn032yzpnckxpyra7-python3.7-setuptools-41.2.0/lib/python3.7/site-packages/setuptools-41.2.0-py3.7.egg/setuptools/__init__.py", line 145, in setup
    File "/nix/store/ljbn79cwlwvrlmqil0nk4zvf043ph592-python3-3.7.5/lib/python3.7/distutils/core.py", line 148, in setup
        dist.run_commands()
    File "/nix/store/ljbn79cwlwvrlmqil0nk4zvf043ph592-python3-3.7.5/lib/python3.7/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
    File "/nix/store/ljbn79cwlwvrlmqil0nk4zvf043ph592-python3-3.7.5/lib/python3.7/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
    File "/nix/store/qg1m9rpl2yvfnjw9r7bh7hx7jijk233a-python3.7-wheel-0.33.6/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 192, in run
        self.run_command('build')
    File "/nix/store/ljbn79cwlwvrlmqil0nk4zvf043ph592-python3-3.7.5/lib/python3.7/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
    File "/nix/store/ljbn79cwlwvrlmqil0nk4zvf043ph592-python3-3.7.5/lib/python3.7/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
    File "/nix/store/ljbn79cwlwvrlmqil0nk4zvf043ph592-python3-3.7.5/lib/python3.7/distutils/command/build.py", line 135, in run
        self.run_command(cmd_name)
    File "/nix/store/ljbn79cwlwvrlmqil0nk4zvf043ph592-python3-3.7.5/lib/python3.7/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
    File "/nix/store/ljbn79cwlwvrlmqil0nk4zvf043ph592-python3-3.7.5/lib/python3.7/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
    File "setup.py", line 51, in run
        from tools.nnwrap import generate_wrappers as generate_nn_wrappers
    ModuleNotFoundError: No module named 'tools.nnwrap'
    builder for '/nix/store/gag7dhx91bzp6i2rxc0ql1cmd78h8kqn-python3.7-torch-0.1.2.post2.drv' failed with exit code 1
    cannot build derivation '/nix/store/dlm02884w3fss89ppcrbli5vm79d7dn1-python3-3.7.5-env.drv': 1 dependencies couldn't be built
    error: build of '/nix/store/dlm02884w3fss89ppcrbli5vm79d7dn1-python3-3.7.5-env.drv' failed

</details>

**Comment by DavHau** [comment]
Mach-nix tries to install a quite old version (0.1.2.post2) of pytorch here. The reason for this is that pytorch switched to wheel releases only from that point on. Mach-nix doesn't support wheels yet. It is my number one focus right now to make them available (check https://github.com/DavHau/mach-nix/issues/7), but currently they are not.

The specific reason why 0.1.2.post2 cannot be installed is because the package is broken. It tries to import `tools.nnwrap` which doesn't exist under tools. The installation also fails with pip under debian.

**Comment by tbenst** [comment]
ah, thanks for the explanation...

I thought Mach was going to take the pytorch derivation from nixpkgs & modify the src/version? I may have the wrong mental model of how this works...

**Comment by DavHau** [comment]
Yes, that is correct. But the list of sources it knows doesn't contain anything better than '0.1.2'. It doesn't consider that in your case it would be best to just take the package from nixpkgs without touching the src attribute.
This could be fixed by adding nixpkgs as a provider to the dependency resolver. But it will also work as soon as wheel support is there.

**Comment by tbenst** [comment]
Ah, I see, thanks!

**Comment by DavHau** [comment]
This should be resolved by version 2.0.0.
Both nixpkgs provider and wheel provider have now been added
